mlir/test/Dialect/Affine/affine-loop-interchange.mlir

```diff
@@ -0,0 +1,264 @@
+// RUN: mlir-opt %s -affine-loop-interchange | FileCheck %s
+
+func @interchange_for_spatial_locality(%A: memref<2048x2048xf64>) {
+  affine.for %i = 0 to 2048 {
+    affine.for %j = 0 to 2048 {
+      %v = affine.load %A[%j, %i] : memref<2048x2048xf64>
+      affine.store %v, %A[%j, %i] : memref<2048x2048xf64>
+    }
+  }
+  return
+}
+
+// Interchanged for spatial locality.
+// CHECK:       affine.load %arg0[%arg1, %arg2] : memref<2048x2048xf64>
+// CHECK-NEXT:  affine.store %{{.*}}, %arg0[%arg1, %arg2] : memref<2048x2048xf64>
```

Comment:
Avoid numbered arguments. Instead capture IVs and use captured values. Here and everywhere below.

Reply:
Fixed. Thanks.

Reply:
Replaced with captured IV variables. Thanks.